vm/src/bytecode.rs

```diff
@@ -260,6 +234,42 @@ pub enum BlockType {
 }
 */
 
+impl CodeObject {
+    pub fn new(
+        arg_names: Vec<String>,
+        varargs: Option<Option<String>>,
+        kwonlyarg_names: Vec<String>,
+        varkeywords: Option<Option<String>>,
+        source_path: String,
+        first_line_number: usize,
+        obj_name: String,
+    ) -> CodeObject {
+        CodeObject {
+            instructions: Vec::new(),
+            label_map: HashMap::new(),
+            locations: Vec::new(),
+            arg_names,
+            varargs,
+            kwonlyarg_names,
+            varkeywords,
+            source_path,
+            first_line_number,
+            obj_name,
+            is_generator: false,
+        }
+    }
+
+    pub fn get_constants<'a>(&'a self) -> impl Iterator<Item = &'a Constant> {
+        self.instructions.iter().filter_map(|x| {
```

Comment:
Hmm, our code object is significantly different from cpython. Should we fake cpython, or expose an incompatible but more truthworthy rustpython codeobject representation to python? This is an open question, I am curious to opinions here.

Reply:
Yeah, I thought about this quite a bit. There's three options:

1. Implement ```dis```, etc. in rust without any attempt to mimic CPython.

* Easy to do.
* Gives us the tools we want now.
* Gives up on supporting any other tools that manipulate the byte code.

2. Implement the compatibility layer to mock out CPython. (More of this.)

* Going to require lots of repetitive, fragile code, or macro dark magic.
* May be hard to get enough compatibility anyway.

3. Make the VM/Bytecode closer to CPython.

* Probably a fair amount of work.
* CPython probably has good performance reasons for its implementation, so we may need to do this eventually anyway.
* Likely to get best compatibility.

Consider ```"a" is "a"``` as a simple example of a different behaviour that is hard to make optimal in the current design.

I'm leaning towards 1 for now, and considering 2/3 again as we start to optimise.

Reply:
In favor of option 1 for now. There might be some bytecode standardization in future, we will adapt at that point. Most importantly, we should have some form of the dis module now.